### DISCUSSION TITLE
Change WIndows Terminal Color Scheme

### DISCUSSION THREAD

**Discussion by xmha97** [AutoDarkMode/Windows-Auto-Night-Mode#781] [Custom scripts]
# Change WIndows Terminal Color Scheme
![84571393-17956880-ada8-11ea-8ec5-4b94c1659324](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/23485114/553acd1d-b710-401b-826c-1aef3639361d)

**Comment by narkissoschan** [comment]
1. Download the attachment and rename it to terminal-color-scheme.ps1
[terminal-color-scheme.txt](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/files/13034578/terminal-color-scheme.txt)
2. open script config and edit it:
Name: Windows Terminal Color Scheme
    Command: powershell    
    WorkingDirectory: %YOUR_SCRIPT_PATH%
    ArgsLight: [%YOUR_SCRIPT_PATH%\terminal-color-scheme.ps1 -settings '%TERMINAL_SETTINGS_FILE%' -colorScheme 'Tango Light']
    ArgsDark: [%YOUR_SCRIPT_PATH%\terminal-color-scheme.ps1 -settings '%TERMINAL_SETTINGS_FILE%' -colorScheme 'Tango Dark']
    AllowedSources: [Any]

**Reply by xmha97** [reply]
Thanks

**Reply by katuwiran** [reply]
Thank you!

**Comment by mi-skam** [comment]
Not working for me

**Comment by norsemangrey** [comment]
I don't think you need a script for this. You can configure any terminal profile to switch scheme based on the OS light/dark mode. You have to do this in the settings.json file as there is not UI for this option yet.

```
"colorScheme": 
{
    "dark": "Solarized Dark",
    "light": "Solarized Light"
},
```

**Reply by kaem-e** [reply]
This worked amazingly, Thanks!

**Reply by muihiuwev** [reply]
Not working. I inserted to `profiles`>`defaults`, it looks like this:

```json
"defaults": 
{
      "colorScheme": 
    {
        "light": "Frost",
        "dark": "Tango Dark"
    },
    "opacity": 50,
    "useAcrylic": true
},
```

What should I do?

**Reply by kaem-e** [reply]
@muihiuwev you need to include it in your profile entry for whatever shell youre using, not the default entry. right next to it there should be a list key where all of your profiles will be, 

```json
    "profiles":
    {
        "defaults": { },
        "list":
        [
            {
                "colorScheme":
                {
                    "dark": "Tomorrow Night",
                    "light": "Tomorrow"
                },
                "commandline": "pwsh.exe -nologo",
                "font":
                {
                    "face": "CaskaydiaCove Nerd Font"
                },
                "guid": "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
                "hidden": false,
                "name": "PowerShell",
                "opacity": 90,
                "padding": "3",
                "scrollbarState": "hidden",
                "source": "Windows.Terminal.PowershellCore",
                "useAcrylic": true
            },
        ]
    }

```

id suggest just searching for the name of your profile, then setting the `colorScheme` key

**Reply by muihiuwev** [reply]
@kaem-e Thank you! It's working now.

**Reply by sterno900** [reply]
~~does not work on stable release~~ 
EDIT: yes it does sorry

**Reply by kaem-e** [reply]
@sterno900 i think it does, im currently on a stable release as well 
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/46230985/7fb3bc21-04df-4e6c-ad91-a299472efd8a)

**Reply by jankalfus** [reply]
Setting it just within the `defaults` property seems to be working now. I'm running Windows Terminal version 1.19.10821.0. Make sure you have "Use Windows theme" set under Settings > Appearance.

**Reply by Miqueas** [reply]
> Setting it just within the defaults property seems to be working now. I'm running Windows Terminal version 1.19.10821.0. Make sure you have "Use Windows theme" set under Settings > Appearance.

I just tried it and yes, it works.

**Reply by vhxnyu** [reply]
Oddly enough, it is not currently (ver1.21.3231.0) built into the terminal settings either😇
